Branch-SDK/src/io/branch/referral/ServerRequest.java

```diff
@@ -161,10 +162,12 @@ public JSONObject getPostWithInstrumentationValues(ConcurrentHashMap<String, Str
         try {
             //Add original parameters
             if (params_ != null) {
-                Iterator<String> keys = params_.keys();
+                String paramsString = params_.toString();
+                JSONObject originalParams = new JSONObject(paramsString);
```

Comment:
minor, but to make this expressible, can this be:

`JSONObject originalParams = new JSONObject(params_.toString());`